src/mappings.py

```diff
@@ -0,0 +1,168 @@
+
+        validation_status_element = external_reference.find("DisorderMappingValidationStatus")
+        if validation_status_element is not None:
+            validation_status = validation_status_element.find("Name").text
+
+        # Check if DisorderMappingICDRelation and DisorderMappingICDRefUri elements exist
+        icd_relation_element = external_reference.find("DisorderMappingICDRelation")
+        if icd_relation_element is not None:
+            icd_relation = icd_relation_element.find("Name").text
+
+        icd_ref_uri_element = external_reference.find("DisorderMappingICDRefUri")
+        if icd_ref_uri_element is not None:
+            icd_ref_uri = icd_ref_uri_element.text
+
+        # Append values to lists
+        orpha_codes.append(orpha_code)
+        names.append(name)
+        sources.append(source)
+        references.append(reference)
+        mapping_relations.append(mapping_relation)
+        validation_statuses.append(validation_status)
+        icd_relations.append(icd_relation)
+        icd_ref_uris.append(icd_ref_uri)
+
+    # Create DataFrame
+    data = {
+        "OrphaCode": orpha_codes,
+        "Name": names,
+        "Source": sources,
+        "Reference": references,
+        "DisorderMappingRelation": mapping_relations,
+        "DisorderMappingValidationStatus": validation_statuses,
+        "DisorderMappingICDRelation": icd_relations,
+        "DisorderMappingICDRefUri": icd_ref_uris
+    }
+    df = pd.DataFrame(data)
+    return df
+
+def run(
+    input_nomenclature_xml: str = INPUT_NOMENCLATURE_XML, input_sssom_config: str = INPUT_CONFIG,
+    outpath: str = OUTPUT_FILE
+):
+    """Run"""
+    # Get data
+    df: pd.DataFrame = xml_as_df(input_nomenclature_xml)
+
+    # Create / rename columns
+    df['predicate_id'] = df['DisorderMappingRelation'].apply(
+        lambda x: 'skos:exactMatch' if x.startswith('E (')
+        else 'skos:narrowMatch' if x.startswith('NTBT (')
+        else 'skos:broadMatch' if x.startswith('BTNT (')
+        else 'skos:relatedMatch')  # 0 instances of relatedMatch; just a fallback
+    df['subject_id'] = df['OrphaCode'].apply(lambda x: 'Orphanet:' + str(x))
+    df['object_id'] = df['DisorderMappingICDRefUri'].apply(lambda x: 'icd11.foundation:' + str(x) if x else None)
+    df = df.rename(columns={'Name': 'subject_label'})
+
+    # Filter by only if ICD11Foundation ref exists && is exact match
+    df = df[(df['DisorderMappingICDRefUri'].notna()) & (df['predicate_id'] == 'skos:exactMatch')]
+
+    # Get only columns we care about
+    df = df[['subject_id', 'subject_label', 'predicate_id', 'object_id']]
+
+    # Write file
+    with open(input_sssom_config, 'r') as yaml_file:
+        metadata: Dict = yaml.load(yaml_file, Loader=yaml.FullLoader)
+    converter = curies.Converter.from_prefix_map(metadata['curie_map'])
+    msdf: MappingSetDataFrame = MappingSetDataFrame(converter=converter, df=df, metadata=metadata)
+    with open(outpath, 'w') as f:
```

Comment:
## What happens to [ordo-icd11.sssom.tsv](https://github.com/monarch-initiative/icd11/releases/download/v2024-03-30/ordo-icd11.sssom.tsv)?
I'm releasing this file in the ICD11 repo, but I don't think I've written anything in `mondo-ingest` or `mondo` to grab it yet. What should we do with it?

Reply:
@matentzn ^^

Reply:
These were only needed as an intermediary to create Mondo mappings. They're nice to have though. Leave them.